functions/src/index.ts

```diff
@@ -73,55 +80,55 @@ app.post("/registerUrl", async (req, res) => {
-    const errorMsg =
+app.get("/go/:id", (req, res) => {
+  const documentId = req.params.id;
+  if (!documentId || documentId.length === 0) {
+    const externalError =
       "Missing URL parameter. " +
-      "Expected structure: https://<...>.net/api/SHORT_URL_HERE";
-    res.status(400).send(errorMsg);
-    return;
+      "Expected structure: https://....net/api/go/SHARE_LINK_ID_HERE";
+    throw new ShareLinkError(400, res, externalError);
   }
 
-  getUrlDocumentDataById(shortUrl)
-    .then((response) => {
-      if (response.data) {
-        const data = response.data;
-        const fullUrl = data.url;
-        const image = data.imageUrl ?? "";
-        const title = data.title ?? "";
-        const description = data.description ?? "";
-        // TODO need to make sure that http-equiv="Refresh" actually allows us to track clicks/get
-        // analytics. See discussion on redirect methods here: https://stackoverflow.com/a/1562539/14034347
-        res.status(200).send(
-          `<!doctype html>
-            <head>
-              <meta http-equiv="Refresh" content="0; url='${fullUrl}'" />
-              <meta property="og:url" content url="${fullUrl}"/>
-              <meta property="og:title" content="${title}"/>
-              <meta property="og:description" content="${description}"/>
-              <meta property="og:image" content="${image}" />
-              <meta name="twitter:card" content="summary_large_image" />
-              <meta property="twitter:title" content="${title}"/>
-              <meta property="twitter:description" content="${description}"/>
-              <meta property="twitter:image" content="${image}"/>
-            </head>
-          </html>`
-        );
+  getUrlDocumentDataByIdStrict(documentId)
+    .then((data) => {
+      const fullUrl = data.url;
+      const image = data.imageUrl ?? "";
+      const title = data.title ?? "";
+      const description = data.description ?? "";
+      // TODO need to make sure that http-equiv="Refresh" actually allows us to track clicks/get
+      // analytics. See discussion on redirect methods here: https://stackoverflow.com/a/1562539/14034347
+      res.status(200).send(
+        `<!doctype html>
+          <head>
+            <meta http-equiv="Refresh" content="0; url='${fullUrl}'" />
+            <meta property="og:url" content url="${fullUrl}"/>
+            <meta property="og:title" content="${title}"/>
+            <meta property="og:description" content="${description}"/>
+            <meta property="og:image" content="${image}" />
+            <meta name="twitter:card" content="summary_large_image" />
+            <meta property="twitter:title" content="${title}"/>
+            <meta property="twitter:description" content="${description}"/>
+            <meta property="twitter:image" content="${image}"/>
+          </head>
+        </html>`
+      );
+    })
+    .catch((error: Error) => {
+      if (error.message === "No share link found") {
```

Comment:
This string matching seems fragile... I think what I was originally imagining is that getUrlDocumentDataByIdStrict() could return a `ShareLinkError` and then we'd just propagate it to the user here...  but I don't feel super strongly about how we do error handling. It was just a suggestion.